tools/opam-single-version-check.bash

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# makes sure no ordinary opam packages have more than a single version
+# the only exceptions are ocaml and xenctrl.
+# This is required because of how the rpm distributions are built: they
+# try to install all versions of all ordinary packages, which fails because
+# the versions of a single package are never compatible.
+
+packages=$(opam list --required-by "xs-toolstack" --recursive --short)
+for pkg in $packages; do
+  the_versions=$(opam info -f all-versions "$pkg")
+  versions=$(echo "$the_versions" | wc -w)
+  # the compiler is allowed to have more than one version because it's a
+  # separate rpm. xenctrl has 2 versions: dummy and master
+  if [ "$versions" -gt 1 ] && echo "$pkg" | grep -vqE '^ocaml$|^ocaml-base-compiler$|^ocaml-system$|^xenctrl$' ; then
```

Comment:
Changed so to match only 4 exact patterns.

Reply:
How about the ppxlib? I think that needs 2 versions. One for 4.08 and one for 4.10, or does the latest version work with both?